### PR TITLE
Release/0.14.2

### PR DESCRIPTION
**Pull request by lmath**

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Comment by lmath** [comment]
Notes on testing: 
1. I created a release candidate of what was on this release branch, 0.14.2-rc3 (rc1 was an outdated version of this branch, and rc2 was created specifically to test an enrich release) 
2. I then pushed two tags, `0.14.2-rc3/gcp/large` and `0.14.2-rc3/aws/large`

### gcp mini
- I brought up a mini in gcp and saw it was the version I expected
image
- I sent through some test events via https://gcp-sandbox-leighan.mini.snplow.net/home#/example-events and saw them come through in Kibana
- I mostly wanted to bring up the mini and see some test events come through, but I configured an [api enrichment](https://gist.github.com/lmath/8915370fcc3a9c3eea94977a6eedb3f7) for fun anyway that disabled the cache, which definitely meant this mini is using enrich 3.2.5
```
curl -v PUT http://<my mini's dns name>/iglu-server/api/schemas -d '{"$schema":"http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#","self":{"vendor":"com.leighan","name":"geo_location","format":"jsonschema","version":"1-0-0"},"description":"geo location","properties":{"country":{"type":"string","maxLength":255,"description":"Mandatory field to describe the country"}},"additionalProperties":true,"type":"object","required":["country"]}' --header 'apikey: <key>'
```
image

### aws mini 
todo

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,11 @@
+Version 0.14.2 (2022-07-08)
+---------------------------
+Bump collector to 2.6.2 (#360)
+Bump Elasticsearch Loader to 1.0.7 (#361)
+Bump Enrich to 3.2.5 (#362)
+Fix links on overview page (#359)
+Update how gpg key is loaded when installing Caddy (#364)
+
```
Comment (oguzhanunlu) [inline CHANGELOG:9]: just a nit: this update shouldn't live in enrich version bump but in release commit

**Comment by oguzhanunlu** [comment]
`Fix synced folder failing on Windows due to lack in NFS support` commit message misses `(close #365)` part

File: CHANGELOG (hunk 1 above)
Comment (lmath) [reply CHANGELOG:9]: Thanks for spotting this! I'll fix it now.

**Review by oguzhanunlu — APPROVED** [review]
Thanks @lmath ! LGTM 👍

**Pull request merged**